### PR TITLE
Revamp README

### PR DESCRIPTION
**Pull request by jonashaag**
cc @pavelzw 

WDYT about deduplicating README further and listing some nice features there. What are some nice Mamba/Micromamba features other than `repoquery` and lock file support @wolfv?

TODO

- [x] Add provision-with-micromamba
- [x] Add lock file installation

### CODE REVIEW

**Review by pavelzw — COMMENTED** [review]
We should unify capitalization... Either conda, mamba and micromamba, `conda`, `mamba` and `micromamba` or Conda, Mamba and Micromamba. 
As from what I've seen, it's mainly `conda`, `mamba` and `micromamba` in the documentation (with some exceptions).

The first sentence

> Mamba is a reimplementation of the conda package manager in C++.

would probably also need to be adjusted to `mamba`. GitHub didn't let me suggest changes on this one though...

**Comment by jonashaag** [comment]
@wolfv do you like the changes?

File: README.md [hunk 1]
```diff
@@ -23,95 +23,58 @@
...
+To efficiently query repositories and query package dependencies you can use `mamba repoquery` or `micromamba repoquery`.
+See the [repoquery documentation](https://mamba.readthedocs.io/en/latest/user_guide/mamba.html#repoquery) for details.
 
-Now you are ready to install packages with
+### Installing lock files
 
-```bash
-mamba install xtensor-r -c conda-forge
-```
+`micromamba` can be used to install lock files generated by (conda-lock)[https://conda-incubator.github.io/conda-lock/] without having to install `conda-lock`.
```
Comment (pavelzw) [inline README.md:52]:
```suggestion
`micromamba` can be used to install lock files generated by [conda-lock](https://conda-incubator.github.io/conda-lock/) without having to install `conda-lock`.
```

**Review by pavelzw — CHANGES_REQUESTED** [review]

**Comment by wolfv** [comment]
Thanks, nice improvement!

@jonashaag if it's OK for you, please don't add branches in this repo (still use your personal fork) :)

Cheers!

**Comment by jonashaag** [comment]
Whoops, seems to be GitHub’s default behavior. Will try to avoid it in the future.

**Pull request merged**